auto-configure fails on POWER9

When trying to build BLIS 0.8.1 on a POWER9 system (hosted by https://osuosl.org) with GCC 10.3 and binutils 2.36.1, I'm seeing the following problem with the `configure` script:

```
$ ./configure --prefix=/tmp/$USER/BLIS-0.8.1 --enable-cblas --enable-threading=openmp --enable-shared auto
configure: detected Linux kernel version 4.18.0-240.1.1.el8_3.ppc64le.
configure: python interpeter search list is: python python3 python2.
configure: using 'python' python interpreter.
configure: found python version 3.9.5 (maj: 3, min: 9, rev: 5).
configure: python 3.9.5 appears to be supported.
configure: C compiler search list is: gcc clang cc.
configure: using 'gcc' C compiler.
configure: C++ compiler search list is: g++ clang++ c++.
configure: using 'g++' C++ compiler (for sandbox only).
configure: found gcc version 10.3.0 (maj: 10, min: 3, rev: 0).
configure: checking for blacklisted configurations due to gcc 10.3.0.
configure: checking gcc 10.3.0 against known consequential version ranges.
configure: found assembler ('as') version 2.36.1 (maj: 2, min: 36, rev: 1).
configure: checking for blacklisted configurations due to as 2.36.1.
configure: warning: assembler ('as' 2.36.1) does not support 'bulldozer'; adding to blacklist.
configure: warning: assembler ('as' 2.36.1) does not support 'sandybridge'; adding to blacklist.
configure: warning: assembler ('as' 2.36.1) does not support 'haswell'; adding to blacklist.
configure: warning: assembler ('as' 2.36.1) does not support 'piledriver'; adding to blacklist.
configure: warning: assembler ('as' 2.36.1) does not support 'steamroller'; adding to blacklist.
configure: warning: assembler ('as' 2.36.1) does not support 'excavator'; adding to blacklist.
configure: warning: assembler ('as' 2.36.1) does not support 'skx'; adding to blacklist.
configure: warning: assembler ('as' 2.36.1) does not support 'knl'; adding to blacklist.
configure: configuration blacklist:
configure:   bulldozer sandybridge haswell piledriver steamroller excavator skx knl
configure: reading configuration registry...done.
configure: determining default version string.
configure: could not find '.git' directory; using unmodified version file.
configure: starting configuration of BLIS 0.8.1.
configure: configuring with official version string.
configure: found shared library .so version '3.0.0'.
configure:   .so major version: 3
configure:   .so minor.build version: 0.0
configure: automatic configuration requested.
/tmp/eb-al3haik0/ccXdk4Z9.o:config_detect.c:function main: error: undefined reference to 'bli_cpuid_query_id'
collect2: error: ld returned 1 exit status
./configure: line 1142: ./auto-detect.x: No such file or directory
configure: hardware detection driver returned ''.
configure: checking configuration against contents of 'config_registry'.
configure: 'auto-detected configuration '' is NOT registered!
configure:
configure: *** Cannot continue with unregistered configuration ''. ***
configure:
```

Is auto-configure expected to work on POWER9 systems?
Maybe something else going wrong here, perhaps something as simple as a missing `#include` statement?

--- a/frame/base/bli_cpuid.c
+++ b/frame/base/bli_cpuid.c
@@ -485,7 +485,7 @@ bool bli_cpuid_is_bulldozer
 	return TRUE;
 }
 
-#elif defined(__aarch64__) || defined(__arm__) || defined(_M_ARM)
+#elif defined(__aarch64__) || defined(__arm__) || defined(_M_ARM) || defined(_ARCH_PPC)
 
 arch_t bli_cpuid_query_id( void )
 {
@@ -530,9 +530,14 @@ arch_t bli_cpuid_query_id( void )
 			return BLIS_ARCH_GENERIC;
 		}
 	}
-	else if ( vendor == VENDOR_UNKNOWN )
+	else if ( vendor == VENDOR_IBM )
 	{
-		return BLIS_ARCH_GENERIC;
+		if ( model == MODEL_POWER7)
+			return BLIS_ARCH_POWER7;
+		else if ( model == MODEL_POWER9)
+			return BLIS_ARCH_POWER9;
+		else if ( model == MODEL_POWER10)
+			return BLIS_ARCH_POWER10;
 	}
 
 	return BLIS_ARCH_GENERIC;
@@ -1203,7 +1208,7 @@ uint32_t bli_cpuid_query
 	return VENDOR_ARM;
 }
 
-#elif defined(__arm__) || defined(_M_ARM)
+#elif defined(__arm__) || defined(_M_ARM) || defined(_ARCH_PPC)
 
 /* 
    I can't easily find documentation to do this as for aarch64, though
@@ -1239,6 +1244,20 @@ uint32_t bli_cpuid_query
 	char  ptno_str[ TEMP_BUFFER_SIZE ];
 	char  feat_str[ TEMP_BUFFER_SIZE ];
 	char* r_val;
+
+#ifdef _ARCH_PPC
+	r_val = find_string_in( "cpu", proc_str, TEMP_BUFFER_SIZE, pci_str );
+	if ( r_val == NULL ) return VENDOR_IBM;
+
+	if ( strstr( proc_str, "POWER7" ) != NULL )
+		*model = MODEL_POWER7;
+	else if ( strstr( proc_str, "POWER9" ) != NULL )
+		*model = MODEL_POWER9;
+	else if ( strstr( proc_str, "POWER10" ) != NULL )
+		*model = MODEL_POWER10;
+
+	return VENDOR_IBM;
+#endif
 
 	//printf( "bli_cpuid_query(): beginning search\n" );
 

--- a/frame/base/bli_cpuid.h
+++ b/frame/base/bli_cpuid.h
@@ -161,19 +161,23 @@ enum
 	FEATURE_AVX512VL = 0x4000
 };
 
-#elif defined(__aarch64__) || defined(__arm__) || defined(_M_ARM)
+#elif defined(__aarch64__) || defined(__arm__) || defined(_M_ARM) || defined(_ARCH_PPC)
 
 char* find_string_in( char* target, char* buffer, size_t buf_len, char* filepath );
 
 enum
 {
 	VENDOR_ARM = 0,
+	VENDOR_IBM,
 	VENDOR_UNKNOWN
 };
 enum
 {
 	MODEL_ARMV7 = 0,
 	MODEL_ARMV8,
+	MODEL_POWER7,
+	MODEL_POWER9,
+	MODEL_POWER10,
 	MODEL_UNKNOWN
 };
 enum

